app/Client/GuzzleAdUserClient.php

```diff
@@ -25,10 +25,13 @@
 use Adshares\Common\Application\Dto\Taxonomy;
 use Adshares\Common\Application\Factory\TaxonomyFactory;
 use Adshares\Common\Application\Service\TargetingOptionsSource;
+use Adshares\Supply\Application\Dto\ImpressionContext;
+use Adshares\Supply\Application\Dto\UserContext;
+use Adshares\Supply\Application\Service\UserContextProvider;
 use GuzzleHttp\Client;
 use function GuzzleHttp\json_decode;
 
-final class GuzzleAdUserClient implements TargetingOptionsSource
+final class GuzzleAdUserClient implements TargetingOptionsSource, UserContextProvider
```

Comment:
GuzzleAdUserClient and DummyAdUserClient have different interface

Reply:
Yes. It is Intentional.